packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -132,12 +135,16 @@ export function SelectResource({
           <Image src={serverIcon} width="32px" mr={2} /> Desktop
         </Flex>
       ),
+      permissionsNeeded: [acl.desktops.read, acl.desktops.list],
     },
   ];
 
-  const initialSelected = tabs.findIndex(
-    component => component.name === location?.state?.entity
+  const index = tabs.findIndex(
+    component => component.name === selectedResource
   );
+  const selectedTabIndex = index > 0 ? index : 0;
```

Comment:
[nit]
There is nifty `Math.max(0, index)` function. I find it easier to read. But that's subjective